src/components/decimal/decimal.test.js

```diff
@@ -0,0 +1,290 @@
+
+import {
+  fieldHelpPreview,
+  getDataElementByValue,
+  tooltipPreview,
+  commonDataElementInputPreview,
+} from "../../../cypress/locators/index";
+
+context("Tests for Decimal component", () => {
+  describe("check props for Decimal component", () => {
+    const input = [
+      [0, "1", "1"],
+      [0, "134^", "134^"],
+      [0, "1234567789", "1,234,567,789"],
+      [0, ",,,,,", ",,,,,"],
+      [0, ".....", "....."],
+      [0, "abc,.123,.!@.00", "abc,.123,.!@.00"],
+      [0, "1,234$", "1,234$"],
+
+      [1, "1", "1.0"],
+      [1, "1.2", "1.2"],
+      [1, "1.23", "1.23"],
+
+      [2, "2", "2.00"],
+      [2, "2.1", "2.10"],
+      [2, "2.12", "2.12"],
+      [2, "2.123", "2.123"],
+
+      [3, "2.1", "2.100"],
+      [3, "2.12", "2.120"],
+      [3, "2.123", "2.123"],
+      [3, "2.1234", "2.1234"],
+
+      [4, "1.1234", "1.1234"],
+      [4, "2", "2.0000"],
+      [4, "234556654", "234,556,654.0000"],
+      [4, "%^%^%<<,,,", "%^%^%<<,,,"],
+
+      [5, "1", "1.00000"],
+      [5, "1.12345", "1.12345"],
+      [5, "1a.23", "1a.23"],
+
+      [6, "2.123456", "2.123456"],
+      [6, "2.1", "2.100000"],
+      [6, "2a.12", "2a.12"],
+      [6, "1,232.123", "1,232.123000"],
+
+      [7, "1", "1.0000000"],
+      [7, "2344.1234567", "2,344.1234567"],
+      [7, "88652344.1234567", "88,652,344.1234567"],
+
+      [8, "1", "1.00000000"],
+      [8, "1.2", "1.20000000"],
+      [8, "1.23", "1.23000000"],
+
+      [9, "2", "2.000000000"],
+      [9, "2.1", "2.100000000"],
+      [9, "1222.12", "1,222.120000000"],
+      [9, "2.123000000", "2.123000000"],
+
+      [10, "2.1", "2.1000000000"],
+      [10, "2.12", "2.1200000000"],
+      [10, "1222.123", "1,222.1230000000"],
+      [10, "2.12345", "2.1234500000"],
+
+      [11, "1", "1.00000000000"],
+      [11, "2345", "2,345.00000000000"],
+      [11, "17899536472345", "17,899,536,472,345.00000000000"],
+
+      [12, "1", "1.000000000000"],
+      [12, "1.12345", "1.123450000000"],
+      [12, "1a.23", "1a.23"],
+
+      [13, "2", "2.0000000000000"],
+      [13, "2.1", "2.1000000000000"],
+      [13, "2a.12", "2a.12"],
+      [13, "1232.123", "1,232.1230000000000"],
+
+      [14, "2.1", "2.10000000000000"],
+      [14, "2.12", "2.12000000000000"],
+      [14, "2.123", "2.12300000000000"],
+      [14, "1222.1234", "1,222.12340000000000"],
+
+      [15, "1", "1.000000000000000"],
+      [15, "2332.78", "2,332.780000000000000"],
+      [15, "1a3.55", "1a3.55"],
+      [15, "1.12345", "1.123450000000000"],
+      [15, "1a.23", "1a.23"],
+    ];
+
+    it.each(input)(
+      "should use %s as precision and %s as input value to produce %s output value",
+      (precision, inputValue, outputValue) => {
+        mount(<Decimal precision={precision} />);
+
+        // then run our tests
+        commonDataElementInputPreview().type(inputValue).blur({ force: true });
+        commonDataElementInputPreview().should("have.value", outputValue);
+        return this;
```

Comment:
I don't think we need to return anything in these it blocks

Reply:
done